typescript/cli/src/rebalancer/factories/RebalancerContextFactory.ts

```diff
@@ -101,6 +102,8 @@ export class RebalancerContextFactory {
       this.warpCore,
       this.metadata,
     );
+
+    return new WithSemaphore(this.config, executor);
```

Comment:
maybe for a following iteration, make `WithSemaphore` inherit `Executor` and build a factory that can instantiate different solutions, and decides what to use based on config or conditions?

Reply:
Felt the decorator pattern more appropriate but we can refactor in the future

Reply:
yes, that why the "maybe", I was just thinking in consistency